core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1754,9 +1754,14 @@ public void removeNode(@Nonnull Node n) throws IOException {
     /**
      * Updates an existing {@link Node} on disk.
      * @see Nodes#updateNode
+     * @since 1.634
      */
     public boolean updateNode(Node n) throws IOException {
-        return nodes.updateNode(n);
+        if ( nodes.updateNode(n) ){
```

Comment:
this should be done in `Nodes.updateNode()` - so it follows the same route as `setNodes()` `addNodes()` and `removeNodes()`